File: src/com/Torvald/Terrarum/Actors/Actor.java
package com.Torvald.Terrarum.Actors;

import org.newdawn.slick.GameContainer;
import org.newdawn.slick.Graphics;

/**
 * Created by minjaesong on 15-12-31.
 */
public interface Actor {

    void update(GameContainer gc, int delta_t);

    long getRefID();
}

File: src/com/Torvald/Terrarum/Actors/Actor.kt
package com.Torvald.Terrarum.Actors

import org.newdawn.slick.GameContainer

/**
 * Created by minjaesong on 16-03-14.
 */
interface Actor {

    fun update(gc: GameContainer, delta_t: Int)

    /**
     * Valid RefID is equal to or greater than 32768.
     * @return Reference ID. (32768-0x7FFF_FFFF_FFFF_FFFF)
     */
    var referenceID: Long?
}